internal/controller/ols_app_redis_reconciliator.go

```diff
@@ -0,0 +1,80 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+)
+
+func (r *OLSConfigReconciler) reconcileRedisServer(ctx context.Context, olsconfig *olsv1alpha1.OLSConfig) error {
+	r.logger.Info("reconcileRedisServer starts")
+	tasks := []ReconcileTask{
+		{
+			Name: "reconcile Redis Deployment",
+			Task: r.reconcileRedisDeployment,
+		},
+		{
+			Name: "reconcile Redis Service",
+			Task: r.reconcileRedisService,
+		},
```

Comment:
@xrajesh I like this idea. But, can we cover that as a part of this follow up story: https://issues.redhat.com/browse/OLS-287 please? Which covers the entire persistent storage use-case. I agree that we have interesting ideas around cache, but adding them one by one in a single PR is violating the rule of "_PRs with atomic commits_" making them big with many changes. Since this PR covers use case for the story attached in the description, can we get this merged and kindly take the rest in followups?